lib/features/vpc/pages/give_vpc_page.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:givt_app/features/vpc/cubit/vpc_cubit.dart';
+import 'package:givt_app/features/vpc/models/vps_response.dart';
+import 'package:givt_app/features/vpc/pages/vpc_intro_page.dart';
+import 'package:givt_app/features/vpc/pages/vpc_success_page.dart';
+import 'package:webview_flutter/webview_flutter.dart';
+
+class GiveVPCPage extends StatelessWidget {
+  const GiveVPCPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: BlocConsumer<VPCCubit, VPCState>(
+        listener: (context, state) {
+          log('vpc state changed on $state');
+          if (state is VPCErrorState || state is VPCCanceledState) {
+            log(state.toString());
+            ScaffoldMessenger.of(context).showSnackBar(
+              SnackBar(
+                content: const Text(
+                  'Cannot get VPC. Please try again later.',
+                  textAlign: TextAlign.center,
+                ),
+                backgroundColor: Theme.of(context).errorColor,
+              ),
+            );
+            context.read<VPCCubit>().showProfiles();
+          }
+        },
+        builder: (context, state) {
+          if (state is VPCProfilesOverview) {
+            return _createProfilesOverviewPage(context);
+          } else if (state is VPCInfoState) {
+            return const VPCIntroPage();
+          } else if (state is VPCFetchingURLState) {
+            return const Center(child: CircularProgressIndicator());
+          } else if (state is VPCSuccessState) {
+            return const VPCSuccessPage();
+            // } else if (state is VPCCanceledState) {
+            //   return _createVPCCanceledPage(context);
+          } else if (state is VPCWebViewState) {
+            return _createWebViewPage(context, state.response);
+          } else {
+            return Container();
+          }
```

Comment:
Why don't you return the `Container` at the end and use a switch statement?

Reply:
could you please elaborate with an example?

Reply:
@pavelpavelko, I upgraded the project to latest version of dart that adds some nice [features](https://codewithandrea.com/articles/whats-new-dart-3-introduction/) 
One of them is the switch 
```suggestion
switch (state) {
    VPCProfilesOverview => _createProfilesOverviewPage(context),
    VPCInfoState => const VPCIntroPage(),
    VPCFetchingURLState => const Center(child: CircularProgressIndicator()),
    VPCSuccessState => const VPCSuccessPage(),
    VPCCanceledState => _createVPCCanceledPage(context),
    VPCWebViewState => _createWebViewPage(context, state.response),
     _ =>Container()
     };
        
```

Reply:
finally! I'm so expecting this feature :-)

Reply:
there is small mistake in the logic (current impl is w/o separate cancel  page). 
but I realy like your idea, will improve it as well later (do not want to update dart right now ;) )